src/Template/Deserialization.php

```diff
@@ -62,7 +62,16 @@ function {{functionName}}(array {{jsonPath}}): {{className}}
 EOT;
 
     private const TMPL_ASSIGN_DATETIME_FROM_FORMAT = <<<'EOT'
-{{modelPath}} = \DateTime::createFromFormat('{{format}}', {{jsonPath}});
+foreach([{{formats|join(', ')}}] as {{format}}) {
```

Comment:
ah i see you still pass parameters for these names. but do we need custom names or could we just use these plain names?
we would need to set `format = false` before looping to avoid leaking previous date fields but otherwise i think we can use the variables.